### PR TITLE
Move gRPC code to its own special place

### PR DESCRIPTION
**Pull request by surfnerd**

### CODE REVIEW

File: UnitySDK/Assets/ML-Agents/Editor/DemonstrationImporter.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public override void OnImportAsset(AssetImportContext ctx)
                 Stream reader = File.OpenRead(ctx.assetPath);
 
                 var metaDataProto = DemonstrationMetaProto.Parser.ParseDelimitedFrom(reader);
-                var metaData = new DemonstrationMetaData(metaDataProto);
+                var metaData = metaDataProto.FromProto();
```
Comment (chriselion) [inline UnitySDK/Assets/ML-Agents/Editor/DemonstrationImporter.cs]: The naming feels weird on this. Guess I'm just used to `Foo.FromBar()` be a static method in `Foo` not an instance method on `Bar`

File: UnitySDK/Assets/ML-Agents/Scripts/Grpc/GrpcExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                NumberEpisodes = dm.numberEpisodes,
+                DemonstrationName = dm.demonstrationName
+            };
+            return demoProto;
+        }
+        
+        /// <summary>
+        /// Initialize metadata values based on proto object.
+        /// </summary>
+        public static DemonstrationMetaData FromProto(this DemonstrationMetaProto demoProto)
```
Comment (chriselion) [inline UnitySDK/Assets/ML-Agents/Scripts/Grpc/GrpcExtensions.cs]: Maybe `ToDemonstrationMetaData`? `ToManaged`? not sure what else to call it though.

File: UnitySDK/Assets/ML-Agents/Editor/DemonstrationImporter.cs (hunk 1 above)
Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Editor/DemonstrationImporter.cs]: It's technically a static method, just looks like an instance method.

Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Editor/DemonstrationImporter.cs]: But I agree with your sentiment.  I just wanted to remove as much Grpc code away from code that doesn't need to know about it

File: UnitySDK/Assets/ML-Agents/Scripts/Grpc/GrpcExtensions.cs (hunk 2 above)
Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Scripts/Grpc/GrpcExtensions.cs]: sounds good.

**Pull request merged**